stdlib/pervasives.ml

```diff
@@ -294,6 +294,17 @@ external open_desc : string -> open_flag list -> int -> int = "caml_sys_open"
 external set_out_channel_name: out_channel -> string -> unit =
   "caml_ml_set_channel_name"
 
+(* [finally_ f g x] calls [f x], but takes care of always calling [g x]
+   whether [f] raises or returns *)
+let finally_ f g x =
+  try
+    let res = f x in
+    g x;
+    res
+  with e ->
+    g x;
+    raise e
+
 let open_out_gen mode perm name =
```

Comment:
Noting the various comments about this below, I realise that the uses of `finally_` *here* shouldn't cause this semantics error, but is there any reason to leave this erroneous version of `finally_` in and not replace it with @nojb's better suggestion (which doesn't try to call `g` twice if `g` raises)?